Assets/LeapMotionModules/UI/Interaction/InteractionBehaviourBase.cs

```diff
@@ -0,0 +1,130 @@
+﻿using Leap;
+using Leap.Unity.Attributes;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace Leap.Unity.UI.Interaction {
+
+  public abstract class InteractionBehaviourBase : MonoBehaviour {
+
+    public InteractionManager interactionManager;
+
+    protected Rigidbody _body;
+    public Rigidbody Rigidbody { get { return _body; } }
+
+    [SerializeField]
+    [HideInInspector]
+    #pragma warning disable 0414
+    // Variable used only for inspector rendering.
+    private bool _interactionManagerIsNull = true;
+    #pragma warning restore 0414
+
+    [Header("Interaction Settings")]
+    [DisableIf("_interactionManagerIsNull", isEqualTo: true)]
+    public bool ignoreHover = false; // TODO: 
+    [DisableIf("_interactionManagerIsNull", isEqualTo: true)]
+    public bool ignoreContact = false; // TODO: Contact NYI.
+    [DisableIf("_interactionManagerIsNull", isEqualTo: true)]
+    public bool ignoreGrasping = false;
+    /// <summary> Can this object be grasped with two or more hands? </summary>
+    [DisableIf("_interactionManagerIsNull", isEqualTo: true)]
+    public bool allowMultiGrasp = false;
+
+    /// <summary>
+    /// Called by the InteractionManager every FixedUpdate, after
+    /// InteractionHands have FixedUpdated.
+    /// 
+    /// InteractionBehaviour uses this to fire per-object interaction events.
+    /// </summary>
+    public abstract void FixedUpdateObject();
+
+    protected virtual void Awake() {
+      _body = GetComponent<Rigidbody>();
+      _body.maxAngularVelocity = 100F;
+
+      if (interactionManager != null) {
+        interactionManager.RegisterInteractionBehaviour(this);
+      }
+    }
+
+    protected virtual void Start() {
```

Comment:
This should probably be done in OnEnable instead of start so that you can register/unregister by enabling/disabling behaviors